user.go

```diff
@@ -39,6 +42,58 @@ func (a *App) GetUser(userID string) (*User, error) {
 	}
 }
 
+// GetUserByIdentifier gets a user using their identifier
+// returns user on success, error on failure
+func (a *App) GetUserByIdentifier(identifier string) (*User, error) {
+	var errorText string
+	message := "failed to get Passage User By Identifier"
+	limit := 1
+	res, err := a.client.ListPaginatedUsersWithResponse(
+		context.Background(),
+		a.ID,
+		&ListPaginatedUsersParams{
+			Limit:      &limit,
+			Identifier: &identifier,
+		},
+	)
+
+	if err != nil {
+		return nil, Error{Message: "network error:failed to get Passage User by Identifier"}
+	}
+
+	if res.JSON200 != nil {
+		users := res.JSON200.Users
+		if len(users) == 0 {
+			message = fmt.Sprintf(IdentifierDoesNotExist, identifier)
+			return nil, Error{
+				Message:    message,
+				StatusCode: 404,
+				StatusText: "404 not found",
+				ErrorText:  "user_not_found",
```

Comment:
Go has status code helpers in [net/http](https://pkg.go.dev/net/http#StatusText).
& the default error text for this code can be found [here](https://github.com/passageidentity/backend-services/blob/b701b7acbda785a7bc3c66fbab6e08b861edafb5/common/management-gateway-service/openapi/main.go#L56).

So this could look like:
```suggestion
				StatusCode: http.StatusNotFound,
				StatusText: fmt.Sprintf("%v %v", http.StatusNotFound, http.StatusText(http.StatusNotFound)),
				ErrorText:  "User not found",
```

Reply:
update this code. Did something very similar.